pkg/driver/common/generator/base_assets.go

```diff
@@ -111,9 +112,15 @@ var (
 			"sidecar.yaml", "common/hypershift/sidecar_add_kubeconfig.yaml.patch",
 		),
 	}
-	// DefaultLivenessProbe is definition of the default livenessprobe sidecar.
-	DefaultLivenessProbe = generator.SidecarConfig{
-		TemplateAssetName: LivenessProbeAssetName,
+	// DefaultLivenessProbeDeployment is definition of the default livenessprobe sidecar for Deployment.
+	DefaultLivenessProbeDeployment = generator.SidecarConfig{
+		TemplateAssetName: LivenessProbeDeploymentAssetName,
+		ExtraArguments:    nil,
+		HasMetricsPort:    false,
+	}
+	// DefaultLivenessProbeDaemonSet is definition of the default livenessprobe sidecar for DaemonSet.
+	DefaultLivenessProbeDaemonSet = generator.SidecarConfig{
```

Comment:
I think `DefaultHostNetworkLivenessProbe` would be a better name - it's not about Deployment/DaemonSet, but about using host network or not. We had control plane pods using host network in the past and I think we had also daemonset using the pod network.

The same would apply to the sidecar patch files - `assets/common/sidecars/livenessprobe_hostnetwork.yaml`